src/syscheckd/create_db.c

```diff
@@ -1574,6 +1626,89 @@ void fim_diff_folder_size(){
     os_free(diff_local);
 }
 
+void update_wildcards_config(){
+    OSListNode *node_it;
+    OSListNode *aux_it;
+    directory_t *dir_it;
+    directory_t *new_entry;
+    char **paths;
+
+    if (syscheck.wildcards == NULL || syscheck.directories == NULL) {
+        return;
+    }
+
+    mdebug2(FIM_WILDCARDS_UPDATE_START);
+    w_rwlock_wrlock(&syscheck.directories_lock);
+
+    OSList_foreach(node_it, syscheck.directories) {
+        dir_it = node_it->data;
+        dir_it->is_expanded = 0;
+    }
+
+    OSList_foreach(node_it, syscheck.wildcards) {
+        dir_it = node_it->data;
+        paths = expand_wildcards(dir_it->path);
+
+        if (paths == NULL) {
+            continue;
+        }
+
+        for (int i = 0; paths[i]; i++) {
+            new_entry = fim_copy_directory(dir_it);
+            os_free(new_entry->path);
+
+            new_entry->path = paths[i];
+#ifndef WIN32
+            if (CHECK_FOLLOW & new_entry->options) {
+                new_entry->symbolic_links = realpath(new_entry->path, NULL);
+            }
+#else
+            str_lowercase(new_entry->path);
+#endif
+            new_entry->is_expanded = 1;
+
+            if (new_entry->diff_size_limit == -1) {
+                new_entry->diff_size_limit = syscheck.file_size_limit;
+            }
+
+            fim_insert_directory(syscheck.directories, new_entry);
+        }
+        os_free(paths);
+    }
+
+    node_it = OSList_GetFirstNode(syscheck.directories);
+    while (node_it != NULL) {
+        dir_it = node_it->data;
+        if (dir_it->is_wildcard && dir_it->is_expanded == 0) {
+            // Send event if needed before delete this node entry
+            fim_process_missing_entry(dir_it->path, FIM_SCHEDULED, NULL);
```

Comment:
Just realized this call might block when sending deleted events while holding the `syscheck.directories_lock` exclusively, also blocking realtime and whodata threads.

We will need to split the update operation:
- Add any new entries to the configuration prior to scanning.
- Remove the unexpanded entries from the global directories and move them to an auxiliar list.
- Use the auxiliar list to trigger the deleted events without using the global directories list, (we will need to write a new function in order to do this).